src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -2000,6 +2004,142 @@ public void testStats() throws DockerException, InterruptedException {
     assertThat(stats.network(), notNullValue());
   }
 
+  @Test
+  public void testNetworks() throws Exception {
+    assumeTrue("Docker API should be at least v1.21 to support Container Creation with " +
+                   "HostConfig, got " + sut.version().apiVersion(),
+               compareVersion(sut.version().apiVersion(), "1.21") >= 0);
+
+    assumeFalse(CIRCLECI);
+
+    final String networkName = randomName();
+    final Ipam ipam =
+        Ipam.builder().driver("default").config("192.168.0.0/24", "192.168.0.0/24", "192.168.0.1")
+            .build();
+    final NetworkConfig networkConfig =
+        NetworkConfig.builder().name(networkName).driver("bridge").checkDuplicate(true).ipam(ipam)
+            .build();
+
+    final NetworkCreation networkCreation = sut.createNetwork(networkConfig);
+    assertThat(networkCreation.id(), is(notNullValue()));
+    assertThat(networkCreation.getWarnings(), is(nullValue()));
+
+    final List<Network> networks = sut.listNetworks();
+    assertTrue(networks.size() > 0);
+
+    Network network = null;
+    for (Network n : networks) {
+      if (n.name().equals(networkName)) {
+        network = n;
+      }
+    }
+    assertThat(network, is(notNullValue()));
+    assertThat(network.id(), is(notNullValue()));
+    assertThat(sut.inspectNetwork(network.id()).name(), is(networkName));
+    assertThat(network.ipam(), equalTo(ipam));
+
+    sut.removeNetwork(network.id());
+
+    try {
```

Comment:
Would using the `@Rule ExpectedException` above be cleaner here? I.e. `exception.expect(NetworkNotFoundException.class)`.